src/main/java/org/opensearch/knn/jni/JNIService.java

```diff
@@ -173,11 +175,13 @@ public static KNNQueryResult[] queryIndex(
         long indexPointer,
         float[] queryVector,
         int k,
+        @Nullable Integer efSearch,
```

Comment:
nit : Can we have nullable either at end of parameter or let's remove nullable , we can null any how With Integer.

Reply:
- Why at the end of the parameter?
- Its a clear indication that this value can be null and modern editors like intellij helps out if you forget the check in code. I would keep it  as it is unless there is a strong reason to remove it.